internal/webhook/kueue/validating.go

```diff
@@ -0,0 +1,260 @@
+//go:build !nowebhook
+
+package kueue
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	componentApi "github.com/opendatahub-io/opendatahub-operator/v2/api/components/v1alpha1"
+	dscv1 "github.com/opendatahub-io/opendatahub-operator/v2/api/datasciencecluster/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/internal/webhook/shared"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/resources"
+)
+
```

Comment:
// Webhooks for Kueue label validation:
// - batch/v1: jobs
// - kubeflow.org/v1: mpijobs, mxjobs, paddlejobs, pytorchjobs, tfjobs, xgboostjobs
// - ray.io/v1alpha1: rayjobs, rayclusters
// - jobset.x-k8s.io/v1alpha2: jobsets